cmd/git-sync/main.go

```diff
@@ -1254,6 +1277,10 @@ func dirIsEmpty(dir string) (bool, error) {
 // removeDirContents iterated the specified dir and removes all contents,
 // except entries which are specifically excepted.
 func removeDirContents(dir string, log *logging.Logger, except ...string) error {
+	return removeDirContentsExceptMostRecent(dir, log, 0, except...)
+}
+
+func removeDirContentsExceptMostRecent(dir string, log *logging.Logger, exceptMoreRecentThan time.Duration, except ...string) error {
```

Comment:
How about a slightly different factoring.  I'm thinking something like:

```
func removeDirContentsIf(dir string, log *logging.Logger, fn func(fi os.FileInfo,) (bool, error) ) error {
```

That would read each dirent, call `Stat`, and call the `fn`.  If `fn` returns error, it's an error.  If it returns true, remove the path.

That way, `removeDirContents` just passes a nil func, and the cleanup path can return a local func which compares the timestamps and filenames.  

I *think* this gets is better locality of logic, what do you think?

Reply:
I agree it seems cleaner, I applied your suggestion